CppImport/src/visitors/ClangAstVisitor.cpp

```diff
@@ -61,12 +62,15 @@ void ClangAstVisitor::setSourceManager(const clang::SourceManager* sourceManager
 	Q_ASSERT(sourceManager);
 	sourceManager_ = sourceManager;
 	trMngr_->setSourceManager(sourceManager);
+	macroImportHelper_.setSourceManager(sourceManager);
 }
 
 void ClangAstVisitor::setPreprocessor(const clang::Preprocessor* preprocessor)
 {
 	Q_ASSERT(preprocessor);
-	preprocessor_ = preprocessor;
+	preprocessor_= preprocessor;
+	macroImportHelper_.setPreprocessor(preprocessor);
+	const_cast<clang::Preprocessor*>(preprocessor)->addPPCallbacks(std::make_unique<PPCallback>(macroImportHelper_));
```

Comment:
This is a bit fishy. Why the const cast? Or more specifically, why receive the preprocessor as const? Is there a better place to add the callbacks class?